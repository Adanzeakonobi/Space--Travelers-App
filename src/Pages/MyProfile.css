.myProfileDisplays{
    display: flex;
    gap: 20px;
    margin-left: 30px;
    margin-right: 30px;
    margin-top: 1.5rem;
}

.myProfileDisplays h3 {
    border: 0.5px solid grey;
    padding: 15px;
}

.myProfileDisplays h4 {
    border: 0.5px solid grey;
    padding: 15px;
}

.reservationDisplay,
.missionsDisplay {
    width: 50%!important;
}

.myProfileDisplays h1 {
    font-size: 30px;
    margin-bottom: 18px;
}