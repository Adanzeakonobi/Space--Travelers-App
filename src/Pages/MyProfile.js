import React from 'react';
import { useSelector } from 'react-redux';
import './MyProfile.css';

const MyProfile = () => {
  const rockets = useSelector((state) => state.rocketsReducer);
  const reservedRockets = rockets.filter((rocket) => rocket.reserv);
  const { missions } = useSelector((state) => state.missions);
  const joinedMissions = missions.filter((missions) => missions.joined);

  return (
    <div className="myProfileDisplays">
      <div className="reservationDisplay">
        <h1>My Rockets</h1>
        {!reservedRockets.length ? (
          <h4 className="NoReservedRockets">No Reserved Rockets</h4>
        ) : (
          reservedRockets.map((rocket) => (
            <h3 key={rocket.rocket_id}>{rocket.name}</h3>
          ))
        )}
      </div>
      <div className="missionsDisplay">
        <h1>My Missions </h1>
        {!joinedMissions.length ? (
          <h3 className="NoReservedRockets">No mission joined</h3>
        ) : (
          joinedMissions.map((joinedOne) => (
            <h3 key={joinedOne.mission_id}>{joinedOne.mission_name}</h3>
          ))
        )}
      </div>
    </div>
  );
};

export default MyProfile;
