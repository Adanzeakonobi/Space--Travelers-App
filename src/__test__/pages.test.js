import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import RocketsPage from '../pages/RocketsPage';
import MissionsPage from '../pages/MissionsPage';
import MyProfile from '../pages/MyProfile';

describe('Testing if all the pages render correctly', () => {
  it('Testing Rockets page (main one)', () => {
    const tree = render(
      <Provider store={store}>
        <RocketsPage />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
  it('Testing if (Missions page) renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <MissionsPage />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
  it('Testing if (My Profile Page) renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <MyProfile />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
