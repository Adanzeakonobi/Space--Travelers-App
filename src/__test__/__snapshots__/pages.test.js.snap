// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing if all pages renders correctly Testing Missions page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="missions"
      >
        <h2>
          Loading..
        </h2>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="missions"
    >
      <h2>
        Loading..
      </h2>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing if all pages renders correctly Testing My Profile Page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="myProfileDisplays"
      >
        <div
          class="reservationDisplay"
        >
          <h1>
            My Rockets
          </h1>
          <h4
            class="NoReservedRockets"
          >
            No Reserved Rockets
          </h4>
        </div>
        <div
          class="missionsDisplay"
        >
          <h1>
            My Missions 
          </h1>
          <h3
            class="NoReservedRockets"
          >
            No mission joined
          </h3>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="myProfileDisplays"
    >
      <div
        class="reservationDisplay"
      >
        <h1>
          My Rockets
        </h1>
        <h4
          class="NoReservedRockets"
        >
          No Reserved Rockets
        </h4>
      </div>
      <div
        class="missionsDisplay"
      >
        <h1>
          My Missions 
        </h1>
        <h3
          class="NoReservedRockets"
        >
          No mission joined
        </h3>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing if all pages renders correctly Testing Rockets page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
